config/sitemap.rb

```diff
@@ -36,28 +36,44 @@
   # add '/static/about_daily', changefreq: 'monthly'
   # add '/static/about_project', changefreq: 'monthly'
   # add '/static/rights', changefreq: 'monthly'
+  ActiveFedora::SolrService.query("+has_model_ssim:Monograph AND -(press_sim:demo OR press_sim:heliotrope OR press_sim:monitoringservicetarget)",
+                                  fl: ['id',
+                                       'date_modified_dtsi',
+                                       'visibility_ssi',
+                                       'tombstone_ssim',
+                                       'representative_id_ssim',
+                                       'file_set_ids_ssim'],
+                                  rows: 100_000).each do |doc|
+    next unless doc['visibility_ssi'] == 'open'
+    next if doc['tombstone_ssim']&.first == 'yes'
 
-  docs = ActiveFedora::SolrService.query("+has_model_ssim:Monograph AND -(press_sim:demo OR press_sim:heliotrope OR press_sim:monitoringservicetarget)", rows: 100_000)
-  docs.each do |d|
-    next unless d['visibility_ssi'] == 'open'
-    next if d['tombstone_ssim']&.first == 'yes'
+    file_set_ids = doc['file_set_ids_ssim']
+    cover_id = doc['representative_id_ssim']&.first
+    rep_file_set_ids = FeaturedRepresentative.where(work_id: doc['id'], file_set_id: file_set_ids).pluck(:file_set_id)
 
-    cover_id = d['representative_id_ssim']&.first
+    file_set_docs = []
+    # We have books with so many file_sets that the query is too big for solr, so we break them up to manage query size if needed
+    if file_set_ids.present?
+      until file_set_ids.empty?
+        file_set_docs << ActiveFedora::SolrService.query("{!terms f=id}#{file_set_ids.shift(999).join(",")}", fl: ['id', 'visibility_ssi'], rows: 1000)
```

Comment:
@sethaj I much prefer getting all of a Monograph's FileSet docs using a single query like [this](https://github.com/mlibrary/heliotrope/blob/30ccfb347e5be7156abcfec2538be64ad026c0bc/lib/tasks/backup_monographs.rake#L86).